api/src/test/java/org/openmrs/api/context/ContextTest.java

```diff
@@ -238,4 +254,111 @@ public void becomeUser_shouldChangeLocaleWhenBecomeAnotherUser() {
 		
 		Context.logout();
 	}
+
+	/**
+	 * @see Context#evictSingleEntity(SessionFactory, Class, String)
+	 */
+	@Test
+	public void evictSingleEntity_shouldClearSingleEntityFromCaches(){
+		//SessionFactory sf = (SessionFactory) applicationContext.getBean("sessionFactory");
+		PersonName name = Context.getPersonService().getPersonName(PERSON_NAME_ID_2);
+		//Load the person so that the names are also stored  in person names collection region
+		Context.getPersonService().getPerson(name.getPerson().getPersonId());
+
+		//Let's have the name in a query cache
+		sf.getCurrentSession().createCriteria(PERSON_NAME_CLASS)
+			.add(Restrictions.eq("personNameId", PERSON_NAME_ID_2))
+			.setCacheable(true)
+			.setCacheRegion(QUERY_REGION)
+			.list();
+		assertTrue(sf.getCache().containsEntity(PERSON_NAME_CLASS, PERSON_NAME_ID_2));
+		assertNotNull(sf.getStatistics().getSecondLevelCacheStatistics(Person.class.getName() + ".names")
+			.getEntries().get(name.getPerson().getPersonId()));
+		assertEquals(1, sf.getStatistics().getCacheRegionStatistics(QUERY_REGION).getElementCountInMemory());
```

Comment:
Same as here @wluyima `sf.getStatistics().getCacheRegionStatistics(QUERY_REGION)` returns `null`.
At this stage, we have not yet called our eviction methods so i think it shouldn't be `null`

Reply:
I think you should be calling sf.getCache().getQueryCache.(QUERY_REGION).getRegion().getElementCountInMemory())

Reply:
@wluyima method `getQueryCache(String)` is not defined in `org.hibernate.Cache` I think the Hibernate version used here has a lot of alternation than that uses in `Omrs:1.9.10` during https://github.com/openmrs/openmrs-module-webservices.rest/commit/5e98bd7cd88832af7b2928f0d1480458d5af1a2c implementation

I think i should go ahead and open a Talk-thread on this.